azavea/raster-vision
# Make max_windows a required argument for RandomWindowGeoDataset

## Overview



This PR makes `max_windows` a required argument for `RandomWindowGeoDataset`, and removes the option for this argument to be `None`. The purpose of this is to avoid a runtime error that would occur when iterating through a PyTorch dataloader that uses this dataset if `max_windows` uses the current default value of `None`. The PR also makes all arguments other than `scene` keyword args in order to allow the order of the args to stay the same.

### Checklist

- [ ] Added `needs-backport` label if the change should be back-ported to the previous release

### Notes


Updated relevant unit tests and tutorials.

## Testing Instructions


See updated unit tests.

Test case (should run with no errors)
```python
from rastervision.core.data import ClassConfig
from rastervision.core.data.utils import make_ss_scene
from rastervision.pytorch_learner import SemanticSegmentationRandomWindowGeoDataset

from torch.utils.data import DataLoader

image_uri = 's3://spacenet-dataset/spacenet/SN7_buildings/train/L15-0331E-1257N_1327_3160_13/images/global_monthly_2018_01_mosaic_L15-0331E-1257N_1327_3160_13.tif'
label_uri = 's3://spacenet-dataset/spacenet/SN7_buildings/train/L15-0331E-1257N_1327_3160_13/labels/global_monthly_2018_01_mosaic_L15-0331E-1257N_1327_3160_13_Buildings.geojson'

class_config = ClassConfig(
    names=['background', 'building'],
    colors=['lightgray', 'darkred'],
    null_class='background')

scene = make_ss_scene(
    class_config=class_config,
    image_uri=image_uri,
    label_vector_uri=label_uri,
    label_vector_default_class_id=class_config.get_class_id('building'),
    image_raster_source_kw=dict(allow_streaming=True))

dataset = SemanticSegmentationRandomWindowGeoDataset(scene, out_size=224, size_lims=[220,224], max_windows=10)

dataloader = DataLoader(dataset, batch_size=5, shuffle=True, num_workers=0)
x, y = next(iter(dataloader))
```
---

Closes #2222
